### PR TITLE
Fixed agent scrolling and resizing issues

### PR DESCRIPTION
**Pull request by abist**
Fixes https://github.com/Microsoft/sqlopsstudio/issues/2064.

### CODE REVIEW

File: src/sql/parts/jobManagement/views/jobStepsView.component.html [hunk 1]
```diff
@@ -22,4 +22,4 @@ <h1 style="display: inline">Steps</h1>
 		</td>
 	</tr>
 </table>
-<div class='steps-tree' #table style="height: 100%; width: 100%"></div>
\ No newline at end of file
+<div class='steps-tree' #table style="height: 485px; width: 100%"></div>
```
Comment (kburtram) [inline src/sql/parts/jobManagement/views/jobStepsView.component.html]: where did the size `485px` come from?  Will this work well on various screen resolutions?

File: src/vs/base/parts/tree/browser/treeViewModel.ts [hunk 2]
```diff
@@ -23,9 +23,10 @@ export class HeightMap {
 		this.indexes = {};
 	}
 
-	public getContentHeight(): number {
+	// {{SQL CARBON EDIT}}
+	public getContentHeight(offsetTop?: number): number {
 		var last = this.heightMap[this.heightMap.length - 1];
-		return !last ? 0 : last.top + last.height;
+		return !last ? 0 : last.top + last.height + offsetTop;
```
Comment (kburtram) [inline src/vs/base/parts/tree/browser/treeViewModel.ts]: does this work correctly when offsetTop is undefined?  I thought adding undefined to a number was `NaN`?

File: src/vs/base/parts/tree/browser/treeView.ts [hunk 3]
```diff
@@ -830,8 +830,10 @@ export class TreeView extends HeightMap {
 	}
 
 	public set scrollTop(scrollTop: number) {
+		// {{SQL CARBON EDIT}}
+		let offset = this.domNode.offsetTop ? this.domNode.offsetTop : 0;
```
Comment (kburtram) [inline src/vs/base/parts/tree/browser/treeView.ts]: what are we trying to do with `offset` here?  How do we know this doesn't impact the behavior of the tree when used in other contexts?

File: src/sql/parts/jobManagement/views/jobStepsView.component.html (hunk 1 above)
Comment (abist) [reply src/sql/parts/jobManagement/views/jobStepsView.component.html]: The 485px is the size from where the steps tree starts to the status bar underneath.

File: src/vs/base/parts/tree/browser/treeView.ts (hunk 3 above)
Comment (kburtram) [reply src/vs/base/parts/tree/browser/treeView.ts]: by "top offset" you mean any tree that isn't located at coordinate `(0,0)` with it's container div?

Comment (abist) [reply src/vs/base/parts/tree/browser/treeView.ts]: @kburtram Yes.

Comment (kburtram) [reply src/vs/base/parts/tree/browser/treeView.ts]: @abist could we solve the problem by including the tree in a div and then applying the top offset to the container?  I'd prefer not to change the shared control class for a particular feature if possible, and I know we're able to position the tree not fixed to the top-left corner in some places such as the Remote File Browser dialog.  Please let me know if I'm missing something, but I still don't understand why we need this change?

screen

Comment (abist) [reply src/vs/base/parts/tree/browser/treeView.ts]: The scrolling works based on the monaco-scrollable-element, whose height's logic doesn't account for offsetTop. So far, this was the only solution I could find, but I'll try other methods where touching vscode side isn't necessary.

Comment (anthonydresser) [reply src/vs/base/parts/tree/browser/treeView.ts]: @abist why does the domnode have an offset? that node is internal to the class and shouldn't be being modified from external sources.

File: src/sql/parts/jobManagement/views/jobStepsView.component.html (hunk 1 above)
Comment (kburtram) [reply src/sql/parts/jobManagement/views/jobStepsView.component.html]: How did you determine this is the ideal number?

**Comment by kburtram** [comment]
@abist where are we with this one?  I still see the edits to the tree control in `vs` directory, so I guess we couldn't find a way to not need that?  Do you think this will be ready for August Public Preview?

**Comment by abist** [comment]
@kburtram I synced up with @anthonydresser on this but couldn't figure it out. I believe he's looking into the issue, without changing vs code.

**Comment by kburtram** [comment]
@abist thanks.  So it sounds like this will miss the August release.

**Comment by abist** [comment]
@kburtram Most likely, unless we have a solution by today that doesn't touch vscode

**Comment by abist** [comment]
Closing this PR since Anthony is looking at the issue.

**Pull request closed without merging**